// ngx-copypaste/__ivy_ngcc__/fesm2015/ngx-copypaste.js
import { Directive, ElementRef, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import * as ɵngcc0 from '@angular/core';
class NgxCopyPasteDirective {
    constructor(_elementRef) {
        this._elementRef = _elementRef;
    }
    copy() {
        let select = window.getSelection();
        if (select) {
            select.removeAllRanges();
        }
        const element = this._elementRef.nativeElement;
        if (element instanceof HTMLInputElement ||
            element instanceof HTMLTextAreaElement) {
            this._elementRef.nativeElement.select();
        }
        else {
            const range = document.createRange();
            range.selectNodeContents(this._elementRef.nativeElement);
            select = window.getSelection();
            if (select) {
                select.addRange(range);
            }
        }
        document.execCommand('copy');
    }
}
NgxCopyPasteDirective.ɵfac = function NgxCopyPasteDirective_Factory(t) { return new (t || NgxCopyPasteDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
NgxCopyPasteDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: NgxCopyPasteDirective, selectors: [["", "ngxCopyPaste", ""]], exportAs: ["copy"] });
NgxCopyPasteDirective.ctorParameters = () => [
    { type: ElementRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxCopyPasteDirective, [{
        type: Directive,
        args: [{
                selector: '[ngxCopyPaste]',
                exportAs: 'copy'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, null); })();

class NgxCopyPasteModule {
}
NgxCopyPasteModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxCopyPasteModule });
NgxCopyPasteModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxCopyPasteModule_Factory(t) { return new (t || NgxCopyPasteModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxCopyPasteModule, { declarations: function () { return [NgxCopyPasteDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [NgxCopyPasteDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxCopyPasteModule, [{
        type: NgModule,
        args: [{
                declarations: [NgxCopyPasteDirective],
                imports: [
                    CommonModule
                ],
                exports: [NgxCopyPasteDirective]
            }]
    }], null, null); })();

/**
 * Generated bundle index. Do not edit.
 */

export { NgxCopyPasteDirective, NgxCopyPasteModule };
