// ngx-copypaste/bundles/ngx-copypaste.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('ngx-copypaste', ['exports', '@angular/core', '@angular/common'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global['ngx-copypaste'] = {}, global.ng.core, global.ng.common));
}(this, (function (exports, core, common) { 'use strict';

    var NgxCopyPasteDirective = /** @class */ (function () {
        function NgxCopyPasteDirective(_elementRef) {
            this._elementRef = _elementRef;
        }
        NgxCopyPasteDirective.prototype.copy = function () {
            var select = window.getSelection();
            if (select) {
                select.removeAllRanges();
            }
            var element = this._elementRef.nativeElement;
            if (element instanceof HTMLInputElement ||
                element instanceof HTMLTextAreaElement) {
                this._elementRef.nativeElement.select();
            }
            else {
                var range = document.createRange();
                range.selectNodeContents(this._elementRef.nativeElement);
                select = window.getSelection();
                if (select) {
                    select.addRange(range);
                }
            }
            document.execCommand('copy');
        };
        return NgxCopyPasteDirective;
    }());
    NgxCopyPasteDirective.decorators = [
        { type: core.Directive, args: [{
                    selector: '[ngxCopyPaste]',
                    exportAs: 'copy'
                },] }
    ];
    NgxCopyPasteDirective.ctorParameters = function () { return [
        { type: core.ElementRef }
    ]; };

    var NgxCopyPasteModule = /** @class */ (function () {
        function NgxCopyPasteModule() {
        }
        return NgxCopyPasteModule;
    }());
    NgxCopyPasteModule.decorators = [
        { type: core.NgModule, args: [{
                    declarations: [NgxCopyPasteDirective],
                    imports: [
                        common.CommonModule
                    ],
                    exports: [NgxCopyPasteDirective]
                },] }
    ];

    /**
     * Generated bundle index. Do not edit.
     */

    exports.NgxCopyPasteDirective = NgxCopyPasteDirective;
    exports.NgxCopyPasteModule = NgxCopyPasteModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
